### PR TITLE
fix: AREAD/AINSERT support in macros called from copybooks

### PR DESCRIPTION
**Pull request by slavek-kucera**
fix: Address edge cases when AREAD and AINSERT are called from source code brought by COPY instruction.
feat: Add support for macro library to fuzzer.

### CODE REVIEW

File: parser_library/test/processing/aread_test.cpp [hunk 1]
```diff
@@ -359,3 +359,385 @@ Line5
...
+public:
+    parse_result parse_library(const std::string& library, analyzing_context ctx, library_data data) override
+    {
+        auto it = m_files.find(library);
+        if (it == m_files.end())
+            return false;
+
+        auto a = std::make_unique<analyzer>(it->second, analyzer_options { library, this, std::move(ctx), data });
+        a->analyze();
+        a->collect_diags();
```
Comment (michalbali256) [inline parser_library/test/processing/aread_test.cpp:377]:
```suggestion
```

Comment (slavek-kucera) [reply parser_library/test/processing/aread_test.cpp:377]: We end up doing this step during normal processing, so while it is redundant here, I'd prefer keeping it.

File: parser_library/src/lsp/text_data_ref_t.cpp [hunk 2]
```diff
@@ -60,6 +58,13 @@ std::string_view text_data_ref_t::get_range_content(range r) const
     return std::string_view(&text->at(start_i), end_i - start_i);
 }
 
+std::string_view text_data_ref_t::get_lines_beginning(position pos) const
```
[the remark this answers is not in the document]
Comment (slavek-kucera) [reply parser_library/src/lsp/text_data_ref_t.cpp:61]: actually, I wanted `get_lines_beginning_at` and `get_line_beginning_at` to have the same signature and did not want to spend time changing the later. So, I'd prefer to keep the position argument for now.

**Review by michalbali256 — APPROVED** [review]
